torch/csrc/autograd/python_function.cpp

```diff
@@ -557,11 +567,19 @@ static void _trace_post_record(
     node = unpacked;
   }
   for (int i = 0; i < num_outputs; ++i) {
-    auto var = (THPVariable*)PyTuple_GET_ITEM(output_objects, i);
-    Value* value = node->outputs()[i];
-    if (var->cdata.defined()) {
-      value->inferTypeFrom(var->cdata);
-      jit::tracer::setValueTrace(var->cdata, value);
+    PyObject* obj = PyTuple_GET_ITEM(output_objects, i);
+    if (THPVariable_Check(obj)) {
+      auto var = (THPVariable*)obj;
+      Value* value = node->outputs()[i];
+      if (var->cdata.defined()) {
+        value->inferTypeFrom(var->cdata);
+        jit::tracer::setValueTrace(var->cdata, value);
+      }
+    } else {
+      // Regular python type.
+      Value* value = node->outputs()[i];
+      value->setType(PyObjectType::create());
+      jit::tracer::setValueTrace(torch::jit::toIValue(py::cast(obj), PyObjectType::create()), value);
```

Comment:
@suo I was wondering if you had some suggestions on how to test JIT is doing the right thing here :)

Reply:
I don't have any special insight except for to do some trace checking tests on representative functions (similar to the ones in `test_jit.py`)